manim/renderer/shader.py

```diff
@@ -33,54 +33,130 @@ def get_shader_code_from_file(file_path):
         return source
 
 
-class Mesh:
-    def __init__(self, shader, attributes, indices=None, use_depth_test=True):
-        self.shader = shader
-        self.attributes = attributes
-        self.indices = indices
-        self.use_depth_test = use_depth_test
-        self.init_updaters()
+def filter_attributes(unfiltered_attributes, attributes):
+    # Construct attributes for only those needed by the shader.
+    filtered_attributes_dtype = []
+    for i, dtype_name in enumerate(unfiltered_attributes.dtype.names):
+        if dtype_name in attributes:
+            filtered_attributes_dtype.append(
+                (
+                    dtype_name,
+                    unfiltered_attributes.dtype[i].subdtype[0].str,
+                    unfiltered_attributes.dtype[i].shape,
+                )
+            )
 
-        self.translation = np.zeros(3)
-        self.rotation = np.zeros(3)
-        self.scale = np.zeros(3)
-        self.model_matrix = np.eye(4)
-        self.model_matrix_needs_update = False
+    filtered_attributes = np.zeros(
+        unfiltered_attributes[unfiltered_attributes.dtype.names[0]].shape[0],
+        dtype=filtered_attributes_dtype,
+    )
 
-    def render(self):
-        # Set matrix uniforms.
-        if self.model_matrix_needs_update:
-            pass
-        try:
-            self.shader.set_uniform(
-                "u_model_matrix", opengl.matrix_to_shader_input(self.model_matrix)
-            )
-        except KeyError:
-            pass
+    for dtype_name in unfiltered_attributes.dtype.names:
+        if dtype_name in attributes:
+            filtered_attributes[dtype_name] = unfiltered_attributes[dtype_name]
 
-        if self.use_depth_test:
-            self.shader.context.enable(moderngl.DEPTH_TEST)
+    return filtered_attributes
 
-        vertex_buffer_object = self.shader.context.buffer(self.attributes.tobytes())
-        if self.indices is None:
-            index_buffer_object = None
-        else:
-            vert_index_data = self.indices.astype("i4").tobytes()
-            if vert_index_data:
-                index_buffer_object = self.shader.context.buffer(vert_index_data)
-            else:
-                index_buffer_object = None
-        vertex_array_object = self.shader.context.simple_vertex_array(
-            self.shader.shader_program,
-            vertex_buffer_object,
-            *self.attributes.dtype.names,
-            index_buffer=index_buffer_object,
-        )
-        vertex_array_object.render(moderngl.TRIANGLES)
-        vertex_buffer_object.release()
-        vertex_array_object.release()
-        if index_buffer_object is not None:
-            index_buffer_object.release()
+
+class Object3D:
```

Comment:
is this an internal structure?

Reply:
No, users can use it too.

Reply:
Ok, so can you create a `__all__` in this file possibly at the top and add things that users should be able to use there?